### ISSUE TITLE
field-surrogate reader: measured performance work before F3's training loop (NFS stat storm, duplicated grid, covering-grid retention)

### ISSUE THREAD

**Issue by eberrigan** [talmolab/mosquito-cfd#112]
Deferred from the #105/#106 review (see the review threads there). All figures below were **measured** by a reviewer against a purpose-built 256×128×256 boxlib plotfile matching the real corpus geometry (513 MB, 67.1 MB/field), not estimated. None of these are correctness problems — the reader is correct — but together they will dominate F3's first corpus iteration.

## 1. `FieldCorpus` does ~553,000 `stat` calls for one corpus pass

Instrumented over a real 27-config manifest with 200 plotfiles per config, iterating `config_ids() → steps() → plotfile()`:

```
5,400 (config, step) pairs addressed
  manifest re-parses                    :     5,428
  Path.iterdir scans                    :     5,427
  Path.is_dir stat calls                :   553,534
  wall time, local SSD, yt never invoked:     43.98 s   (8.14 ms per lookup)
  one load_manifest_configs()           :      0.337 ms
```

Two independent causes:

- **`_configs()` re-reads, re-parses and re-validates the manifest on every public call** (`config_ids`, `_config`, `run_dir`, `steps`, `global_params`, `plotfile`, `snapshot`). Locally 1.8 s; on cluster NFS at 1–5 ms/open, 5–27 s of pure re-reading. The docstring already treats the manifest as static, so caching (optionally with an mtime check) is straightforward.
- **`plotfile()` linearly scans `iterdir()`** with a regex and an `is_dir()` per entry instead of hitting `run / f"plt{step:05d}"` directly — ~100 stats per lookup on a 200-plotfile run dir. A direct hit with fallback-to-scan preserves the variable-digit-width tolerance at one stat.

`Path.iterdir()` also discards the `DirEntry`, throwing away `scandir`'s `d_type` hint; `os.scandir()` + `entry.is_dir()` on the `DirEntry` removes essentially all of these syscalls.

**On cluster NFS — the documented deployment — 553 K stats at 1–5 ms is 9–46 minutes of pure path resolution before a single byte of field data is read.**

## 2. `to_domino_volume`'s `grid` duplicates `volume_mesh_centers` for nothing

Verified numerically at 256×128×256:

```
grid == coords.reshape(nx, ny, nz, 3)  ->  True
reshape is a free view (shares_memory) ->  True
cost of the second meshgrid + stack    ->  +201 MB allocated, 404 MB peak, 0.14 s
```

`np.stack(np.meshgrid(x, y, z, indexing="ij"), axis=-1)` is definitionally the buffer the point cloud already built. `cloud.coords.reshape(nx, ny, nz, 3)` is exactly equivalent and free (`coords` is C-contiguous and read-only).

Secondary, and worth resolving with the encoder design: in stock DoMINO `grid` is the small fixed-resolution SDF bounding grid (O(10⁵) points), **not** the full volume mesh. Emitting an 8.4 M-point `grid` may be semantically wrong as well as expensive.

## 3. `cg.field_data` retains every field for the life of the covering grid

`cg[field]` caches into `cg.field_data`, and `cg` lives for the whole loop, so all six unyt arrays (403 MB) accumulate *on top of* the six `to_ndarray()` copies. Measured with `del cg.field_data[field]` after each unwrap:

```
no clear : t=3.27 s  peak=1006 MB  retained=535 MB
clear    : t=2.92 s  peak= 601 MB  retained= 63 MB
```

−40 % peak, −88 % retained, and marginally **faster**. Two lines in the hot loop of the repo's only plotfile reader; `flow_video` calls it once per frame.

## 4. Sub-box covering grids are ~10× cheaper and bit-identical

For a 32³ near-field box on the 256×128×256 file:

```
sub-box covering_grid, 6 fields : 0.80 s, peak Δ  131 MB
full-grid read then slice       : 4.10 s, peak Δ 1278 MB
results identical               : True (exact)
```

The spec currently pins "the read performed as a full level-0 covering grid sliced in memory" for legacy parity, so this needs a spec change, not just a code change. The legacy docstring's stated reason for the full read (yt's ghost-cell boundary check) applies to `smoothed_covering_grid`/`ds.region`, not `covering_grid`. **Re-verify on a real multi-FAB wing plotfile before acting** — the measurement above used a single-FAB file.

`flow_video`'s `lev-3d` mode calls this once per frame; over a 200-frame video that is ~11 minutes and ~1 GB/frame.

## 5. Smaller items

- `cell_volume` is 67 MB of a repeated scalar. `np.broadcast_to(v, (n,))` is a zero-copy, already-read-only view and satisfies the documented "exists so CC-F3 can vary it later" purpose; swap to `np.full` only when it actually varies.
- `to_point_cloud` keeps `gx/gy/gz` alive until return; `del` after building `coords` returns 201 MB of peak.
- Zero-cell regions (`upper_edge`, `outside_domain`) still construct the full covering grid and read all six fields — **1.28 GB and ~4 s to return nothing**. A short-circuit after the AMR and missing-field guards would make it free (order matters: those guards' tests assert they fire before any read).

## Why this matters now

End-to-end, one full-domain DoMINO example costs ≈**2.2 GB peak / ≈1.3 GB retained** (1278 MB read + 874 MB point cloud + 404 MB grid). An F3 dataloader with even two workers will not fit that on the 24 GB A5000 alongside a model. Worth resolving before F3's training loop is written rather than after.

Advances #20. Context: #105, #106.
